### PR TITLE
Add python source distribution

### PR DESCRIPTION
**Pull request by zozzz**
This helps installing on linux to which not have pre built wheel.

### CODE REVIEW

File: python/main/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["maturin>=0.13,<0.14"]
```
Comment (adamchainz) [inline python/main/pyproject.toml:2]: Is there a specifc reason to add an upper version bound? Generally they [do more harm than good](https://iscinumpy.dev/post/bound-version-constraints/) - I would prefer to keep this open so the project isn’t stuck on an old maturin.

**Review by adamchainz — APPROVED** [review]
Thanks! This will also help with using the package on new/beta Python versions.

File: python/main/pyproject.toml (hunk 1 above)
Comment (zozzz) [reply python/main/pyproject.toml:2]: No reason, i will remove it.

Comment (adamchainz) [reply python/main/pyproject.toml]: Cheers

**Comment by wilsonzlin** [comment]
Thanks @zozzz and @adamchainz!

**Pull request merged**